sma_set.go

```diff
@@ -0,0 +1,50 @@
+package movavg
+
+// SMASet is a set of related calculators of Simple Moving Average:
```

Comment:
Hmmm...

You know, I've examined the code...

It doesn't look like we need such concrete implementation - I mean, bound to `SMA` struct.

I suggest to make a generic implementation to be used like:

```
set := movavg.NewSet(
  movavg.NewSMA(1 * time.Minute),
  movavg.NewSMA(30 * time.Minute),
  movavg.NewSMA(1 * time.Hour),
  // variadic, just to make it nicer
)
set.Add(1)
set.Add(2)
set.Add(3)
for _, avg := range set.Avg() {
  println(avg)
}
```

It's generic, it's looking nice, it simplifies code.

Reply:
Interesting idea. Perhaps do this in addition to what I've got. Below is my use case for initializing my metrics. If the only available way is what you suggest above, then my use case gets very verbose.

```
var SMAWindows = []int{10, 100, 1000, 10000, 100000, 1000000}

m.cycleTime = movavg.NewSMASet(SMAWindows)
m.maintTime = movavg.NewSMASet(SMAWindows)
m.maintTimePercent = movavg.NewSMASet(SMAWindows)
m.genBlockTime = movavg.NewSMASet(SMAWindows)
m.addBlockTime = movavg.NewSMASet(SMAWindows)
m.confBlockTime = movavg.NewSMASet(SMAWindows)
m.evalTime = movavg.NewSMASet(SMAWindows)
m.computedProcTime = movavg.NewSMASet(SMAWindows)
m.computedProcTimePercent = movavg.NewSMASet(SMAWindows)
m.actualProcTimePercent = movavg.NewSMASet(SMAWindows)
m.actualProcTime = movavg.NewSMASet(SMAWindows)
m.blockQCount = movavg.NewSMASet(SMAWindows)
```

Reply:
`type Set []MA` is a problem because I need functions to get the []float64 of all my windows in the set... I'll play around with this and see what I come up with.